AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/SQLStatement+Condition.swift

```diff
@@ -23,30 +23,41 @@ private func translateQueryPredicate(from modelSchema: ModelSchema,
                                      namespace: Substring? = nil) -> SQLPredicate {
     var sql: [String] = []
     var bindings: [Binding?] = []
-    var groupType: QueryPredicateGroupType = .and
     let indentPrefix = "  "
     var indentSize = 1
     var groupOpened = false
 
-    func translate(_ pred: QueryPredicate) {
+    func translate(_ pred: QueryPredicate, opAhead: Bool, groupType: QueryPredicateGroupType) {
```

Comment:
- `opAhead`: decide whether `and` or `or` operator should be added before bracket depends on

- `groupType`: the operation applied to two `QueryPredicateOperation`.

Reply:
if you have to explain the variable name here then consider renaming the variable and refactor as you try to rename it. If the variable still needs some explanation, then better add some documentation to the top of the function

Reply:
Looking at the android implementation, it looks pretty elegant:
https://github.com/aws-amplify/amplify-android/blob/fef0cc31b50c427c60e144346a896c20a2f440e4/aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/adapter/SQLPredicate.java#L108

Maybe we can get some inspiration from their codebase?